cmd/flush_handles/flush_handles.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+
+	"github.com/google/go-tpm-tools/tpm2tools"
+	"github.com/google/go-tpm/tpm2"
+)
+
+var (
+	tpmPath            = flag.String("tpm-path", "/dev/tpm0", "Path to a TPM character device or socket.")
+	flushTransient     = flag.Bool("flush-transient", true, "Flush all transient handles.")
+	flushLoadedSession = flag.Bool("flush-loaded-session", false, "Flush all loaded session handles.")
+	flushSavedSession  = flag.Bool("flush-saved-session", false, "Flush all saved session handles.")
+	flushAllTypes      = flag.Bool("flush-all-types", false, "Flush handles of all handle types. If enabled, settings for other handle types are ignored.")
+)
+
+func main() {
+	flag.Parse()
+	handleTypes := handleTypesFromFlags()
+	err := openTPMAndFlushHandlesOfTypes(handleTypes)
+	if err != nil {
+		log.Fatalf("%v", err)
+	}
+	log.Println("Handles flushed!")
+}
+
+// handleTypeFromFlags parses flag values and returns a slice of
+// tpm2.HandleType enums which should be flushed from the TPM.
+func handleTypesFromFlags() []tpm2.HandleType {
+	if *flushAllTypes {
+		return []tpm2.HandleType{tpm2.HandleTypeTransient, tpm2.HandleTypeLoadedSession, tpm2.HandleTypeSavedSession}
+	}
+	types := []tpm2.HandleType{}
+	if *flushTransient {
+		types = append(types, tpm2.HandleTypeTransient)
+	}
+	if *flushLoadedSession {
+		types = append(types, tpm2.HandleTypeLoadedSession)
+	}
+	if *flushSavedSession {
+		types = append(types, tpm2.HandleTypeSavedSession)
+	}
+	return types
+}
+
+// openTPMAndFlushHandlesOfTypes opens the TPM defined at the flag tpm-path
+// and calls flushHandlesOfType on every type within handleTypes.
+func openTPMAndFlushHandlesOfTypes(handleTypes []tpm2.HandleType) error {
```

Comment:
optional: these func names look too verbose to me. I'd go with
```
func flushAll(types []tpm2.HandleType) error
func flush(rw io.ReadWriter, ht tpm2.HandleType) error
```

Reply:
Done.